client/src/main/java/com/graphhopper/api/GraphHopperGeocoder.java

```diff
@@ -0,0 +1,73 @@
+package com.graphhopper.api;
+
+import com.graphhopper.PathWrapper;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Robin Boldt
+ */
+public class GraphHopperGeocoder
+{
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private OkHttpClient downloader;
+    private String serviceUrl;
+    private String key = "";
+
+    public GraphHopperGeocoder() {
+        this("https://graphhopper.com/api/1/geocode");
+    }
+
+    public GraphHopperGeocoder(String serviceUrl) {
+        this.serviceUrl = serviceUrl;
+        downloader = new OkHttpClient.Builder().
+                connectTimeout(5, TimeUnit.SECONDS).
+                readTimeout(5, TimeUnit.SECONDS).
+                build();
+    }
+
+    public GHGResponse geocode(String search, String locale) {
```

Comment:
Here I would introduce a GHGRequest to avoid expanding the parameter list later further and breaking the API again and again